tools/spice/spice_utils.py

```
@@ -0,0 +1,159 @@
+        """Load a kernel file into the SPICE system."""
+        self.clear()
+        files_loaded = []
+
+        for filename in os.listdir(self.kernel_path):
+            if filename.endswith(tuple(self.extensions)):
+                filepath = os.path.join(self.kernel_path, filename)
+                spice.furnsh(filepath)
+                files_loaded.append(filename)
+            else:
+                logger.info(f"Invalid kernel_type extension: {filename}.")
+
+        return files_loaded
+
+    def clear(self):
+        """Clear the SPICE kernel pool."""
+        logger.info("Clearing SPICE kernel pool.")
+        spice.kclear()
+
+
+def ls_kernels(extensions=None) -> list:
+    """List furnished spice kernels, optionally filtered by specific extensions.
+
+    Parameters
+    ----------
+    extensions: list of str, optional
+        Extensions to filter the kernels. If None, list all kernels.
+
+    Returns
+    -------
+    : list
+        A list of kernel filenames.
+    """
+    count = spice.ktotal("ALL")
+    result = []
+
+    for i in range(count):
+        file, _, _, _ = spice.kdata(i, "ALL")
+        # Append the file if no specific extensions are provided
+        if extensions is None:
+            result.append(file)
+        # Check if the file ends with any of the specified extensions
+        elif any(file.endswith(ext) for ext in extensions):
+            result.append(file)
+
+    return result
+
+
+def ls_spice_constants() -> dict:
+    """List all constants in the Spice constant pool.
+
+    Returns
+    -------
+    : dict
+        Dictionary of kernel constants
+    """
+    try:
+        # retrieve names of kernel variables
+        # * means matches all variables
+        # 0 : starting from first match
+        # 1000 : num variable names retrieved in call
+        # 81 : max len of variable name
+        kervars = spice.gnpool("*", 0, 1000, 81)
+    except NotFoundError:  # Happens if there are no constants in the pool
```

Isn't it worth knowing this as a user so we should just let it raise in this situation?